### PR TITLE
[FLAG-943] Tropical Tree Cover analysis malfunctioning

### PR DESCRIPTION
**Pull request by willian-viana**
## Overview

Tree Cover Density widget doesn't work correctly on Analysis when threshold is added as parameter. 
This widget has its own threshold (>10 as default), but when a user selects Tropical Tree Cover layer and changes the threshold, a new parameter was added to the query, making the request fail.

We did a major refactor at the `getWHEREQuery` to improve readability and add some new validations.

## Demo

Screenshot 2023-10-17 at 15 45 47

Screenshot 2023-09-07 at 17 49 47

## Testing

- Open the map
- Select Tropical Tree Cover
- Change the threshold
- Try to see Tree Cover Density Widget (it should work normally)

### CODE REVIEW

File: services/analysis-cached.js [hunk 1]
```diff
@@ -305,99 +313,123 @@ const getLocationSelect = ({
...
+
+  if (parameterKey === 'threshold') {
+    if (dataset === 'tropicalTreeCover') {
+      paramKey = 'wri_tropical_tree_cover__decile';
+    } else {
+      paramKey = 'umd_tree_cover_density_2000__threshold';
+    }
+  }
+
+  return paramKey;
```
Comment (wri7tno) [inline services/analysis-cached.js:337]: cool, way better to move all that logic inside its own function.

File: services/analysis-cached.js [hunk 2]
```diff
@@ -85,6 +85,14 @@ const SQL_QUERIES = {
...
 const ALLOWED_PARAMS = {
   annual: ['adm0', 'adm1', 'adm2', 'threshold', 'forestType', 'landCategory'],
+  treeCoverDensity: [
+    'adm0',
+    'adm1',
+    'adm2',
+    'threshold',
+    'forestType',
+    'landCategory',
+  ],
```
Comment (wri7tno) [inline services/analysis-cached.js:95]: honest question, `treeCoverDensity` is a new param in the query...? or what is the purpose of adding this array here? trying to understand a bit more this logic.

File: services/analysis-cached.js [hunk 3]
```diff
@@ -305,99 +313,123 @@ const getLocationSelect = ({
...
+   */
+  if (isTreeCoverDensity && paramKeys.includes('threshold')) {
+    paramKeys = paramKeys.filter((item) => item !== 'threshold');
+  }
+
+  paramKeys.forEach((parameter, index) => {
+    const isLastParameter = paramKeys.length - 1 === index;
+    const hasFilterOption = ['forestType', 'landCategory'].includes(parameter);
+    const value = hasFilterOption ? 1 : params[parameter];
+    const polynameMeta = allFilterOptions.find(
```
Comment (wri7tno) [inline services/analysis-cached.js:373]: so, we still don't know what `polyName` refers to 🤷‍♂️

**Review by wri7tno — APPROVED** [review]

File: services/analysis-cached.js (hunk 3 above)
Comment (willian-viana) [reply services/analysis-cached.js:373]: I removed "polyname" variables but not all of them. Polynames for this method were filterOptions from `landCategory` or `forestType`. A weird name for a weird variable. We can remove all references of it later.

File: services/analysis-cached.js (hunk 2 above)
Comment (willian-viana) [reply services/analysis-cached.js:95]: Anytime we need a different value in the dataset argument, we need to create a different object on `AllowedParams` object. As I needed to validate treeCoverDensity dataset, I've created this object.

File: services/__tests__/get-where-query.spec.js [hunk 4]
```diff
@@ -0,0 +1,41 @@
+import { getWHEREQuery } from '../get-where-query';
+
+describe('getWHEREQuery', () => {
```
Comment (wri7tno) [inline services/__tests__/get-where-query.spec.js:3]: 🎉

**Pull request merged**